Nix support

ursi opened the pull request:
This adds a flake with a nix shell for using cargo and running the example that generates the circuit image, and packages for builds of each workspace.

DarthPJB reviewed (changes requested):
I'd suggest removing the .gitignore changes from this PR; and given the nature of the flake, you may wish to squash these changes (as an optional bonus).

ursi committed:
Add nix support

This adds a flake with a nix shell for using cargo and running the
example that generates the circuit image, and packages for builds of
each workspace.
